packages/admin-stories/src/admin/table/TableWithFilterBar.tsx

```diff
@@ -0,0 +1,171 @@
+query users(
+    $pathFunction: any
+    $username: String
+    $name: String
+    $email: String
+    $website: String
+) {
+    users(
+        email: $email
+        username: $username
+        name: $name
+        website: $website
+    ) @rest(type: "User", pathBuilder: $pathFunction) {
+        id
+        name
+        username
+        email
+        phone
+        website
+    }
+}
+`;
+function pathFunction({ args }: { args: { [key: string]: any } }) {
+    interface IPathMapping {
+        [arg: string]: string;
+    }
+    const paramMapping: IPathMapping = {
+        email: "email",
+        username: "username",
+        name: "name",
+        website: "website",
+    };
+
+    const q = Object.keys(args).reduce((acc: { [key: string]: any }, key: string): { [key: string]: any } => {
+        if (paramMapping[key] && args[key]) {
+            acc[paramMapping[key]] = args[key];
+        }
+        return acc;
+    }, {});
+
+    return `users?${qs.stringify(q, { arrayFormat: "brackets" })}`;
+}
+
+interface IQueryData {
+    users: Array<{
+        id: number;
+        name: string;
+        username: string;
+        email: string;
+        phone: string;
+        website: string;
+    }>;
+}
+
+interface IFilterValues {
+    username: any;
+    name: any;
+    email: any;
+    website: any;
+}
+
+interface IVariables extends IFilterValues {
+    pathFunction: any;
+}
+
+const ExampleWithSelect: React.FC = () => {
+    const options = [
+        { value: "Sincere@april.biz", label: "Sincere@april.biz" },
+        { value: "Shanna@melissa.tv", label: "Shanna@melissa.tv" },
+        { value: "Nathan@yesenia.net", label: "Nathan@yesenia.net" },
+    ];
+    return <Field name="email" type="text" component={FinalFormReactSelectStaticOptions} fullWidth options={options} />;
+};
+
+function Story() {
+    const filterApi = useTableQueryFilter<Partial<IFilterValues>>({});
+
+    const { tableData, api, loading, error } = useTableQuery<IQueryData, Partial<IVariables>>()(query, {
+        variables: {
+            email: filterApi.current.email?.email,
+            name: filterApi.current.name?.name,
+            username: filterApi.current.username?.username,
+            website: filterApi.current.website?.url,
+            pathFunction,
+        },
+        resolveTableData: (data) => ({
+            data: data.users,
+            totalCount: data.users.length,
+        }),
+    });
+
+    return (
+        <TableQuery api={api} loading={loading} error={error}>
+            <TableFilterFinalForm filterApi={filterApi}>
+                <Typography variant="h5">FilterBar</Typography>
+                <FilterBar>
+                    <FilterBarField label={"Username"} name="username">
+                        <Field name="username" type="text" component={FinalFormInput} fullWidth />
+                    </FilterBarField>
+                    <FilterBarMoreFilters>
+                        <FilterBarField label={"Email"} name="email">
```

Comment:
Should we name it `FilterBarFilter`, as it's not a `Field` directly, it just wrap (one or multiple) fields?

Or better `FilterBarPopoverFilter` - as that is basically what it does? (that could look nice besides eg. a search-field that is not in a popover)

Reply:
the Name FilterBarPopoverFilter is a good name but i think i had this name before and at some point @johnnyomair mentioned maybe field would be a better name for it without exposing too much technical knowledge with the prefix of popover :) 
But i agree with the opinion of what you said about that Filter instead of Field would describe it better :)

Reply:
I'd argue that this behaviour is not internal, but something that the dev decides for (alternative is to add a field direclty)

@johnnyomair what do you think about `FilterBarPopoverFilter` ?

Reply:
The naming is fine for me. However, as a dev, I'd also expect a component `FilterBarFilter`, in which I can render a basic filter, without the popover. Maybe we can add such a component if we need it.

Reply:
It's definitely a good idea to remove "Field" from the name.